var/spack/repos/builtin/packages/apex/package.py

```diff
@@ -4,30 +4,78 @@
+    # Disable some default dependencies on Darwin/OSX
+    darwin_default = False
+    if sys.platform != 'darwin':
+        darwin_default = True
+
+    # Enable by default
+    variant('activeharmony', default=True, description='Enables Active Harmony support')
+    variant('binutils', default=True, description='Enables Binutils support')
+    variant('otf2', default=True, description='Enables OTF2 support')
+    variant('openmp', default=darwin_default, description='Enables OpenMP support')
+    variant('papi', default=darwin_default, description='Enables PAPI support')
+    variant('gperftools', default=False, description='Enables Google PerfTools TCMalloc support')
+    variant('jemalloc', default=False, description='Enables JEMalloc support')
+
+    # Disable by default
+    variant('cuda', default=False, description='Enables CUDA support')
+    variant('boost', default=False, description='Enables Boost support')
+
+    # Dependencies
+    depends_on('cmake')
+    depends_on('binutils+libiberty+headers', when='+binutils')
+    depends_on('activeharmony@4.6:', when='+activeharmony')
+    depends_on('otf2', when='+otf2')
+    depends_on('gperftools', when='+gperftools')
+    depends_on('jemalloc', when='+jemalloc')
+    depends_on('papi', when='+papi')
+    depends_on('cuda', when='+cuda')
+    depends_on('boost@1.54:', when='+boost')
+
+    # Conflicts
+    conflicts('+jemalloc', when='+gperftools')
 
     def cmake_args(self):
+        args = []
         spec = self.spec
-        return [
-            '-DBOOST_ROOT=%s' % spec['boost'].prefix,
-            '-DUSE_BFD=TRUE',
-            '-DBFD_ROOT=%s' % spec['binutils'].prefix,
-            '-DUSE_ACTIVEHARMONY=TRUE',
-            '-DACTIVEHARMONY_ROOT=%s' % spec['activeharmony'].prefix,
-            '-DUSE_OMPT=TRUE',
-            '-DOMPT_ROOT=%s' % spec['ompt-openmp'].prefix,
-        ]
+        if '+cuda' in spec:
+            args.append('-DAPEX_WITH_CUDA=TRUE')
+        if '+binutils' in spec:
+            args.append('-DBFD_ROOT={0}'.format(spec['binutils'].prefix))
+            args.append('-DUSE_BFD=TRUE')
+        if '+activeharmony' in spec:
+            args.append('-DACTIVEHARMONY_ROOT={0}'.format(
+                spec['activeharmony'].prefix))
+            args.append('-DUSE_ACTIVEHARMONY=TRUE')
+        if '+otf2' in spec:
+            args.append('-DOTF2_ROOT={0}'.format(spec['otf2'].prefix))
+            args.append('-DUSE_OTF2=TRUE')
+        if '+openmp' in spec:
+            args.append('-DUSE_OMPT=TRUE')
+        if '+gperftools' in spec:
+            args.append('-DGPERFTOOLS_ROOT={0}'.format(
+                spec['gperftools'].prefix))
+            args.append('-DUSE_TCMALLOC=TRUE')
+        if '+jemalloc' in spec:
+            args.append('-DJEMALLOC_ROOT={0}'.format(spec['jemalloc'].prefix))
+            args.append('-DUSE_JEMALLOC=TRUE')
+        if '+boost' in spec:
+            args.append('-DBOOST_ROOT={0}'.format(spec['boost'].prefix))
```

Comment:
I would explicitly disable these in an else-clause whenever possible.

Reply:
will do.  thanks for the review!